Feature/24 DASHBOARD VIEW

LupusNocti opened the pull request:
- Added dashboard to the admin.
- Added seeder for the admin.
- Added a feature where can log in using the name.

LupusNocti committed:
Created database for the admin

LupusNocti committed:
created a view for the admin home dashboard

LupusNocti committed:
fixed the routes

LupusNocti committed:
added a seeder file for an admin user

LupusNocti committed:
Can log in using User's name or Admin's name

LupusNocti committed:
added dashboard graph

LupusNocti committed:
added admin dashboard

LupusNocti committed:
removed styles within the blade file

ChronoDevs commented on app/Http/Controllers/AdminController.php:
```diff
@@ -0,0 +1,22 @@
+
+class AdminController extends Controller
+{
+    public function home()
```
add comment above method

ChronoDevs commented on app/Http/Controllers/AdminController.php:
```diff
@@ -0,0 +1,22 @@
+        return view('admin.home');
+    }
+
+    public function users()
```
add comment above method

ChronoDevs commented on app/Http/Controllers/AdminController.php:
```diff
@@ -0,0 +1,22 @@
+        $users = User::all(); // Retrieve all registered users
+        return view('admin.users', compact('users'));
+    }
+
```
remove space

ChronoDevs commented on app/Http/Controllers/AdminController.php:
```diff
@@ -0,0 +1,22 @@
+
+namespace App\Http\Controllers;
+
+use Illuminate\Http\Request;
```
unused
use Illuminate\Http\Request;
use App\Models\Category;

ChronoDevs commented on app/Http/Controllers/Auth/LoginController.php:
```diff
@@ -5,113 +5,48 @@
-     * @return string
-     */
-    public function redirectTo()
+    public function showLoginForm()
```
add comment above method

ChronoDevs commented on app/Http/Controllers/Auth/LoginController.php:
```diff
@@ -5,113 +5,48 @@
-     * @return \Symfony\Component\HttpFoundation\RedirectResponse
-     */
-    public function redirectToFacebook()
+    protected function guard()
```
duplicate return

ChronoDevs commented on app/Http/Controllers/Auth/LoginController.php:
```diff
@@ -5,113 +5,48 @@
-     */
-    public function handleGoogleCallback()
+    
+    protected function authenticated(Request $request, $user)
```
add comment above method

ChronoDevs commented on app/Http/Controllers/Auth/LoginController.php:
```diff
@@ -5,113 +5,48 @@
 use App\Http\Controllers\Controller;
 use App\Providers\RouteServiceProvider;
```
unused

ChronoDevs commented on app/Http/Controllers/Auth/LoginController.php:
```diff
@@ -5,113 +5,48 @@
     }
-}
+    
+    protected function username()
```
add comment above method

ChronoDevs commented on app/Http/Controllers/Auth/LoginController.php:
```diff
@@ -5,113 +5,48 @@
+    
+        return $fieldType;
+    }
+}
```
add last break line

ChronoDevs commented on database/migrations/2023_08_24_221419_create_admin_table.php:
```diff
@@ -0,0 +1,34 @@
+    {
+        Schema::dropIfExists('admin');
+    }
+}
```
add last break line

ChronoDevs commented on package-lock.json:
delete package-lock.json

ChronoDevs commented on routes/web.php:
```diff
@@ -34,3 +34,17 @@
+    Route::get('/admin/categories', [AdminController::class, 'categories'])->name('admin.categories');
+    Route::get('/admin/tags', [AdminController::class, 'tags'])->name('admin.tags');
+
+    Route::post('/admin/categories/create', 'App\Http\Controllers\AdminController@createCategory')->name('admin.categories.create');});
```
please group correctly

ChronoDevs commented on routes/web.php:
```diff
@@ -34,3 +34,17 @@
+
+Route::middleware(['auth:admin'])->group(function () {
+    // Routes only accessible to admins
+    Route::get('/admin/home', [\App\Http\Controllers\AdminController::class, 'home'])->name('admin.home');
```
Please uniform when calling controller path

ChronoDevs replied:
same as [LoginController::class, 'redirectToFacebook']

ChronoDevs replied:
This is unused
web.php
use App\Http\Controllers\Auth\LoginController;
use App\Http\Controllers\Auth\RegisterController;
use App\Http\Controllers\HomeController;

ChronoDevs replied:
also category route please change

ChronoDevs commented on public/js/adminchart.min.js:
```diff
@@ -0,0 +1,120 @@
+      },
+    },
+  },
+});
```
add last break line

ChronoDevs commented on public/js/adminchart.min.js:
```diff
@@ -0,0 +1,120 @@
+//admin chart
+var ctx2 = document.getElementById("chart-line").getContext("2d");
```
use single quote

ChronoDevs commented on database/seeders/AdminsTableSeeder.php:
```diff
@@ -0,0 +1,26 @@
+namespace Database\Seeders;
+use Illuminate\Support\Facades\DB;
+use Illuminate\Support\Facades\Hash;
+
```
remove space

ChronoDevs commented on database/migrations/2014_10_12_000000_create_users_table.php:
```diff
@@ -26,11 +26,10 @@ public function up()
             $table->integer('zip_code')->nullable();
             $table->text('address')->nullable();
             $table->string('email')->unique();
-            $table->string('provider_id')->nullable();
```
Please make sure removing db columns cannot affect on login user side

ChronoDevs commented on app/Models/Admin.php:
```diff
@@ -0,0 +1,32 @@
+
+    public function isAdmin()
+    {
+        return $this->role === 'admin';
```
make an enum
ex

const ADMIN = 'admin';

usage:
return $this->role === self::ADMIN;

ChronoDevs commented on app/Http/Middleware/AdminMiddleware.php:
```diff
@@ -0,0 +1,36 @@
+
+class AdminMiddleware
+{
+    public function handle(Request $request, Closure $next)
```
add comment above method

LupusNocti committed:
Addressed changes to be made in the PR

LupusNocti commented:
- Added necessary comments in respective methods in both controllers and middlewares.
- Created enum for the Admin Model
- Restores the deleted column in the Users table
- Added additional breakline in the respective files

ChronoDevs commented on routes/web.php:
```diff
@@ -34,3 +34,22 @@
 
 Route::post('/logout', [LoginController::class, 'logout'])->name('logout');
+
+Route::middleware(['auth:admin'])->group(function () {
```
group the middleware auth:admin as one

ChronoDevs replied:
line 38 and line 47

LupusNocti committed:
grouped the auth:admin as one

ChronoDevs commented:
Please fix conflict

LupusNocti committed:
Merge branch 'develop' into feature/24_DASHBOARD_VIEW

LupusNocti committed:
Updated the web.php

uniformed the calling of controllers

LupusNocti commented:
Updated the web.php with the requested changes

LupusNocti committed:
Updated web.php
